plugin/core/signature_help.py

```diff
@@ -115,7 +116,8 @@ def _render_label(self, view: sublime.View, signature: SignatureInformation) ->
                     # route relies on the client being smart enough to figure where the parameter is inside of
                     # the signature label. The above case where the label is a tuple of (start, end) positions is much
                     # more robust.
-                    start = label[prev:].find(rawlabel)
+                    label_match = re.search(r"\b{}\b".format(rawlabel), label[prev:])
+                    start = label_match.start() if label_match else -1
```

Comment:
Does it work with

```
def f(str: str, int: int) -> str:
    return ""
```

Reply:
What about `f(x: str, str: int)`

Reply:
> What about `f(x: str, str: int)`

Then I assume it will do the same as before, i.e. mark the first occurence of "str" it can find. But I guess there is no way to fix that without adding complex and language-dependent code about what should be considered as a parameter and what is a type, etc.

But if you want to find an example where this PR would actually be a regression compared to the status quo, you can take this one:

```julia
f(🐰, 🐺) = 🐰 + 🐺
```

Before (at least the underline is visible for the highlighted parameter):
![before](https://user-images.githubusercontent.com/6579999/192588268-73a1f271-a3ed-4a14-9240-54263bb4ec5d.png)

After:
![after](https://user-images.githubusercontent.com/6579999/192588303-504a7c92-8eb1-4e1a-80a7-40e8faff4006.png)

This happens because this PR assumes that the parameter names must always begin and end with what Python's regex considers as a "word" character (otherwise the word boundary `\b` wouldn't match).

But it's probably a very rare example (but not a complete joke example, actually it's adopted with some adjustments from https://www.youtube.com/watch?v=QwVO0Xh2Hbg&t=4765s), which seems to be acceptable in exchange for the improvement from this PR. I also tried with greek letters (which would be much more common than emojis in parameter identifiers), and in that case it works fine because they are considered as word characters.

Reply:
Offtopic :D 
This is the first time seeing emojis being used in code like this XD `f(🐰, 🐺) = 🐰 + 🐺`

Julia ls takes forever(maybe around 30s ~ 1min) to initialize on linux on my machine, is that normal?

Reply:
>This is the first time seeing emojis being used in code like this XD `f(🐰, 🐺) = 🐰 + 🐺`

Yeah, I think nobody would actually use emojis for things like this in real code ;)
It's more that Julia is often used for numerical calculations from a mathematical or physics background, and so the language allows a wide range of unicode characters in identifiers to make possible e.g. the use of greek letters for variables / physical constants or mathematical symbols for special operators.
 
> Julia ls takes forever(maybe around 30s ~ 1min) to initialize on linux on my machine, is that normal?

Hm, 1min seems pretty long, but yes it takes a while for the server to initialize (I can measure ~ 20sec here). That is because the server will be compiled each time when it starts, and also you can observe that using any feature (e.g. hover) will also lag a few seconds when it is used the very first time (again, the particular code parts for that feature will be compiled just-in-time directly before usage). But after that it should be fast. Unfortunately, afaik there is no good way to fully compile Julia code (i.e. the language server) ahead of time yet. But I think it still starts a bit faster than on VSCode :)